lib/ethdev/rte_ethdev_netns.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "rte_ethdev.h"
+
+inline int rte_ethdev_enter_netns(int ns)
+{
+	int origNs = open("/proc/thread-self/ns/net", O_RDONLY | O_CLOEXEC);
+	if (origNs >= 0) {
+		setns(ns,0);
+	}
+	return origNs;
+}
+
+inline int rte_ethdev_switch_netns(char * netns, int * deviceNS, int * originalNS)
+{
+	char ns[128];
+
+	if (netns[0] == '\0')
+		return -1;
+
+	snprintf(ns, 128, "/var/run/netns/%s", netns);
+	*deviceNS = open(ns, O_RDONLY | O_CLOEXEC);
+	*originalNS = rte_ethdev_enter_netns(*deviceNS);
+	return 0;
+}
+
+inline void rte_ethdev_exit_netns(int current_netns, int intended_netns)
```

Comment:
more like switch than exit?

Reply:
why void? why not check the return value of setns
On success, setns() returns 0.  On failure, -1 is returned and
       [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to indicate the error.

Reply:
Fixed.